src/util-decode-mime.c

```diff
@@ -1844,8 +1845,11 @@ static uint8_t * FindMimeHeaderToken(MimeDecField *field, const char *search_sta
     fptr = FindBuffer(field->value, field->value_len, (const uint8_t *)search_start, strlen(search_start));
     if (fptr != NULL) {
         fptr += strlen(search_start); /* Start at end of start string */
-        tok = GetToken(fptr, field->value_len - (fptr - field->value), search_end,
-                &tptr, tlen);
+        uint32_t search_len = field->value_len - (fptr - field->value);
+        if (max_len) {
+            search_len = MIN((int)search_len, max_len);
```

Comment:
I suppose by casting `search_len` to `int` it would become negative if `(fptr - field->value)` is > `field->value_len`. So then we would pass this on but use it as a `uint32_t` leaving the `max_len` parameter ineffective?

Reply:
`FindBuffer` won't search more than `field->value_len` bytes starting at `field->value`. Thus, `fptr - field->value` should never exceed `field->value_len`.

Reply:
My point is that this *should* logic is irrelevant if we make the check more robust. If then for some unforeseen reason our assumption turns out false (possibly due to some future change), we still have an effective check to reduce the effect.